test/unit/installer.test.ts

```diff
@@ -0,0 +1,152 @@
+import { mock, MockProxy } from 'jest-mock-extended'
+import { Installer, DownloadService, GitHubDownloadMeta } from '@jbrunton/gha-installer'
+import { ActionsCore, ActionsToolCache, FileSystem } from '@jbrunton/gha-installer/lib/interfaces'
+import { ReposListReleasesItem } from '@jbrunton/gha-installer/lib/octokit'
+import { K14sReleasesService } from '../../src/k14s_releases_service'
+import { TestOctokit, createTestOctokit } from '../fixtures/test_octokit'
+
+const assetNames = {
+  linux: "ytt-linux-amd64",
+  win32: "ytt-windows-amd64.exe"
+}
+const downloadUrls = {
+  linux: "https://example.com/k14s/ytt/releases/download/0.28.0/ytt-linux-amd64",
+  win32: "https://example.com/k14s/ytt/releases/download/0.28.0/ytt-windows-amd64.exe"
+}
+const downloadPaths = {
+  linux: "/downloads/ytt-linux-amd64",
+  win32: "/downloads/ytt-windows-amd64.exe"
+}
+const binPaths = {
+  linux: "/bin/ytt",
+  win32: "/bin/ytt.exe"
+}
+const expectedContent = "foo bar baz"
+const expectedChecksums = {
+  linux: '"dbd318c1c462aee872f41109a4dfd3048871a03dedd0fe0e757ced57dad6f2d7  ./ytt-linux-amd64"',
+  win32: '"dbd318c1c462aee872f41109a4dfd3048871a03dedd0fe0e757ced57dad6f2d7  ./ytt-windows-amd64.exe"'
+}
+
+describe('Installer', () => {
+  let octokit: TestOctokit
+  let core: MockProxy<ActionsCore>
+  let cache: MockProxy<ActionsToolCache>
+  let fs: MockProxy<FileSystem>
+
+  beforeEach(() => {
+    core = mock<ActionsCore>()
+    cache = mock<ActionsToolCache>()
+    fs = mock<FileSystem>()   
+    octokit = createTestOctokit()
+    octokit.stubListReleasesResponse({ owner: 'k14s', repo: 'ytt' }, [
+      releaseJsonFor('ytt', '0.10.1'), // a more recent security patch
```

Comment:
The order of stubbing here shows that we lookup the latest version correctly. The ordering itself is done in the `gha-installer` repo [here](https://github.com/jbrunton/gha-installer/blob/fa374dd21734b9c1572e3200665023f3e543a55e/src/github_releases_service.ts#L102) (tested [here](https://github.com/jbrunton/gha-installer/blob/fa374dd21734b9c1572e3200665023f3e543a55e/test/github_releases_service.test.ts#L122)).